src/registry.cpp

```diff
@@ -35,15 +35,27 @@
 	#endif
 #endif
 
-std::string Registry::ReadStrValue(HKEY hkey, std::string const& key, std::string const& val) {
+std::string Registry::ReadStrValue(HKEY hkey, std::string const& key, std::string const& val, REGVIEW view = NATIVE) {
 	char value[1024];
 	DWORD size = 1024;
 	DWORD type = REG_SZ;
 	HKEY key_handle;
+	REGSAM desired_access = KEY_QUERY_VALUE;
+
+	switch (view) {
+		case KEY32:
+			desired_access |= KEY_WOW64_32KEY;
+			break;
```

Comment:
Yeah sorry I didn't check the other code which always invokes with KEY32. Nevermind.